### PR TITLE
TK-432-TimeEntries limit 8hrs a day

### PR DESCRIPTION
**Pull request by GuillaumeDavy**
We can now add and edit timeentries for 8hrs a day

### CODE REVIEW

File: frontend/src/utils/utils.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  const hourDisabled = (hour) => {
+    if(entryDuration){
+      return parseInt(hour) + (parseInt(numberOfHoursForDay) - entryDuration) > 8;
+    }
+    return parseInt(hour) + parseInt(numberOfHoursForDay) > 8;
+  };
+  return _.range(1, 9, 1).map(i => <Option key={`option-hour-${i}`} disabled={hourDisabled(i)} value={i} >{i}</Option>);
+};
+
+//Return the className of disabled radio button in EventForm
```
Comment (tkuy) [inline frontend/src/utils/utils.js]: If this concerns EventForm, this shouldn't be in this utils file.

File: frontend/src/utils/utils.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+export const radioDisabledClassName = (disabled) => {
+  if(disabled) {
+    return 'tk-radio-disabled';
```
Comment (tkuy) [inline frontend/src/utils/utils.js]: A radio disabled style should be defined by the less file. (CSS or Less).

File: frontend/src/utils/utils.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import {Select} from 'antd';
+var _ = require('lodash');
+
+const {Option} = Select;
+
+//Returns the options list of hours for a day with some of them may be disabled to be less than 8hrs of timeEntry for a day
+export const hoursOptions = (numberOfHoursForDay, entryDuration) => {
```
Comment (tkuy) [inline frontend/src/utils/utils.js]: Why don't you do a component with this logic ?

**Review by tkuy — CHANGES_REQUESTED** [review]

File: frontend/src/utils/utils.js (hunk 2 above)
Comment (GuillaumeDavy) [reply frontend/src/utils/utils.js]: It already does

Comment (tkuy) [reply frontend/src/utils/utils.js]: No, I mean you're not supposed to check if the radio button is disabled. It must be antd or your css that use the right style.
CSS : 
```
button[disabled]{
  border: 1px solid #999999;
  background-color: #cccccc;
  color: #666666;
}
```
Also Antd probably have a variable to set to change the style when the button is disabled.

File: frontend/src/components/TimeEntry/AddEntryForm.js [hunk 4]
```diff
@@ -180,7 +180,7 @@ const AddEntryForm = ({date, form, timeSheets, onSuccess, onCancel, numberOfHour
                           <Select
                             showSearch
                           >
-                            {hoursOptions(numberOfHoursForDay)}
+                            {HoursOptions(numberOfHoursForDay)}
```
Comment (tkuy) [inline frontend/src/components/TimeEntry/AddEntryForm.js]: It's a component, call it like this : `<HoursOptions ...={...}>`

File: frontend/src/components/TimeEntry/HoursOptions.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import {Select} from 'antd';
+import PropTypes from 'prop-types';
+var _ = require('lodash');
```
Comment (tkuy) [inline frontend/src/components/TimeEntry/HoursOptions.js]: avoid var, use const instead (or let if it needs to be mutable)

File: frontend/src/components/TimeEntry/HoursOptions.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+const {Option} = Select;
+
+const HoursOptions = (numberOfHoursForDay, entryDuration) => {
```
Comment (tkuy) [inline frontend/src/components/TimeEntry/HoursOptions.js]: It should be with Select with options

**Review by tkuy — CHANGES_REQUESTED** [review]

File: frontend/src/components/TimeEntry/SelectHoursComponent.js [hunk 7]
```diff
@@ -15,24 +15,32 @@
...
+const SelectHoursComponent = ({numberOfHoursForDay, entryDuration}) => {
   const hourDisabled = (hour) => {
     if(entryDuration){
       return parseInt(hour) + (parseInt(numberOfHoursForDay) - entryDuration) > 8;
     }
     return parseInt(hour) + parseInt(numberOfHoursForDay) > 8;
   };
-  return _.range(1, 9, 1).map(i => <Option key={`option-hour-${i}`} disabled={hourDisabled(i)} value={i} >{i}</Option>);
+  return (
+    <Form.Item name="numberHours" label="Number of hours:" rules={[{required: true}]}>
```
Comment (tkuy) [inline frontend/src/components/TimeEntry/SelectHoursComponent.js:32]: This shouldn't be in this component, it belongs to the form. The `Form.Item` must be defined by the form and not your sub component.

**Review by tkuy — CHANGES_REQUESTED** [review]

**Review by tkuy — APPROVED** [review]

**Pull request merged**